cli/lsp/language_server.rs

```diff
@@ -168,7 +168,7 @@ impl Inner {
   /// Only searches already cached assets and documents for a line index.  If
   /// the line index cannot be found, `None` is returned.
   fn get_line_index_sync(
-    &self,
+    &mut self,
     specifier: &ModuleSpecifier,
   ) -> Option<LineIndex> {
     let mark = self.performance.mark("get_line_index_sync");
```

Comment:
It seems odd that a method which is only retrieving information "get_line_index_sync" would need to be marked as mutable. 

The mut requirement arrises from the performance counters - which are basically an optional diagnostic utility. They could just as well be in a static global.

Reply:
We could remove the performance counter from this block. While it is somewhat informative, its performance is always caught in an outer function, and if we were concerned about it we could measure performance at the client sites instead.

Reply:
> The mut requirement arrises from the performance counters

It's because of `self.sources`, which mutates itself when you call its `get_line_index()` method (computes and caches the result lazily.)

FWIW, I considered doing `s/get_/compute_/g`, to indicate they're not simple getters but 🤷.

Reply:
Ah, yeah... `source.get_line_index` is not idempotent.  `compute_` does make more sense.  Would be glad to change that down the road too if we don't deal with it in this PR.